services/QuillLMS/app/views/pages/shared/_footer.html.erb

```diff
@@ -1,126 +1,129 @@
-          <img class="lazyload" src="https://assets.quill.org/images/homepage/google_logo.svg" alt='Google.org Logo'/>
-          <img class="lazyload" src="https://assets.quill.org/images/homepage/footer-att.svg" alt='AT&T Logo'/>
-          <img class="lazyload" src="https://assets.quill.org/images/homepage/footer-fast-forward.svg" alt='Fast Forward Logo'/>
-          <img class="lazyload" src="https://assets.quill.org/images/homepage/footer-nbc.svg" alt='NBC Logo'/>
-          <img class="lazyload" src="https://assets.quill.org/images/homepage/footer-robin-hood.svg" alt='Robinhood Logo'/>
+  <% if current_user&.role == 'student' || current_user&.role == 'staff' %>
+    <footer class="student-footer">
+      <div class="inner-container">
+        <div class="privacy-and-terms">
+          <p>© <%= Time.new.year %> Quill</p>
+          <a href="/tos">Terms</a>
+          <a href="/privacy">Privacy</a>
         </div>
-      </a>
-    </div>
-    <div class="privacy-and-social flex-row space-between vertically-centered">
-      <div class="pas-left flex-row">
-        <p>© <%= Time.new.year %> Quill</p>
-        <p>Quill is 501(c)(3) nonprofit organization</p>
-        <a href="/tos">Terms</a>
-        <a href="/privacy">Privacy</a>
-        <a>hello@quill.org</a>
-        <a href="tel:510-671-0222">510-671-0222</a>
       </div>
-      <div class="pas-right flex-row">
-        <a rel="noopener noreferrer" target="_blank" href="http://twitter.com/Quill_org"><i class="fab fa-twitter" aria-hidden="true"></i></a>
-        <a rel="noopener noreferrer" target="_blank" href="https://www.youtube.com/channel/UCG9mgvktG6vxp7XoYEJTiaw/videos?flow=list&view=0&sort=dd&live_view=500"><i class="fab fa-youtube" aria-hidden="true"></i></a>
-        <a rel="noopener noreferrer" target="_blank" href="https://facebook.com/quill.org"><i class="fab fa-facebook" aria-hidden="true"></i></a>
-        <a rel="noopener noreferrer" target="_blank" href="http://instagram.com/quill_org/"><i class="fab fa-instagram" aria-hidden="true"></i></a>
+    </footer>
+  <% else %>
+    <%= render partial: 'application/premium_footer' %>
+    <footer class="<%=footer_class%>">
+      <div class="footer-navigation">
+        <%= render partial: 'pages/shared/teacher_dashboard_footer' if current_user %>
+        <ul>
+          <li>
+            <p>LEARNING TOOLS</p>
+          </li>
+          <li>
+            <a href="/tools/connect">Quill Connect</a>
+          </li>
+          <li>
+            <a href="/tools/lessons">Quill Lessons</a>
+          </li>
+          <li>
+            <a href="/tools/diagnostic">Quill Diagnostic</a>
+          </li>
+          <li>
+            <a href="/tools/proofreader">Quill Proofreader</a>
+          </li>
+          <li>
+            <a href="/tools/grammar">Quill Grammar</a>
+          </li>
+          <li>
+            <a href="/tools/evidence">Quill Reading for Evidence</a>
+          </li>
+        </ul>
```

Comment:
I don't know if it would be an improvement, but did you consider putting all of these blocks into partials the way we did with `explore_activities_footer` and so on?  Just to keep this file stripped down?  Could see that going either way.

Reply:
Agree with @anathomical. I think it would be even cleaner to declare a partial that's just something like `list_of_links` that can just accept an array of hashes with urls and names, and then iterate through to make all of those `<li><a>{...}</a></li>` tags, since we use them a lot throughout these footers.

Reply:
good call, I'll just make a shared `list_of_links_with_header` partial and add a helper class for the link data